src/main/java/frc/robot/commands/CrocodileDefaultCommand.java

```diff
@@ -1,35 +1,56 @@
 package frc.robot.commands;
 
 import java.util.function.BooleanSupplier;
+import java.util.function.DoubleConsumer;
 import java.util.function.DoubleSupplier;
 
+import edu.wpi.first.math.filter.Debouncer;
+import edu.wpi.first.math.filter.Debouncer.DebounceType;
+import edu.wpi.first.wpilibj.Timer;
 import edu.wpi.first.wpilibj2.command.CommandBase;
 import frc.robot.subsystems.CrocodileSubsystem;
 
 public class CrocodileDefaultCommand extends CommandBase {
     private final CrocodileSubsystem subsystem;
-    private final DoubleSupplier motorSpeed;
-    private final BooleanSupplier slowMovement;
+    private final DoubleSupplier trigger;
+    private boolean beamBreakTracker = false;
+    private final Debouncer debouncer;
+    private final Timer timer;
+    DoubleConsumer giveHapticFeedback;
 
-    public CrocodileDefaultCommand(CrocodileSubsystem subsystem, DoubleSupplier motorSpeed,
-            BooleanSupplier slowMovement) {
+    public CrocodileDefaultCommand(CrocodileSubsystem subsystem, DoubleSupplier trigger, DoubleConsumer giveHapticFeedback) {
```

Comment:
Update callers to pass the new arg. I don't see that above.

Reply:
yep, forgot to do that. In regards to the rumble command above, I want to only do a single 0.5 second rumble once a game piece is gotten, which we make sure is true using the debouncer. if that is the case, we set beam break tracker to true, which only becomes false again once we lose the gamepiece. I'm not entirely sure how the debouncer works, but basically i want that last if statement to run only if we don't detect anything for 500 ms. At that point, the tracker is set to false, signifying that we don't have a game piece, and that we are free to rumble the next time something is detected for more than 500 ms.